### PR TITLE
chore: switch from aws-sdk-go v1 to v2

### PR DESCRIPTION
**Pull request by krancour**
I mistakenly used aws-sdk-go v1 in #2037

v2 is preferable because it helps make shorter work of #1885

### CODE REVIEW

**Comment by krancour** [comment]
@hiddeco I thought this merged on 5/24, but apparently not. I had auto-merge enabled and something must have blocked it...

Fast forward to today and I pushed new stuff to this branch thinking that I'd make a _new_ PR from this branch... but nope. Everything passed and auto-merge of this PR included the latest changes meant for a new one.

🤦‍♂️ 

If you wouldn't mind still reviewing 4e6f73e499903b392280b365775c9da56f4e4f14, I can address any feedback you may have in a follow-up.

Sorry about the mix-up.

File: internal/credentials/ecr/pod_identity.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/ecr"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/patrickmn/go-cache"
+
+	"github.com/akuity/kargo/internal/logging"
+)
+
+var ecrURLRegex = regexp.MustCompile(`^([0-9]{12})\.dkr\.ecr\.(.+)\.amazonaws\.com`)
```
Comment (hiddeco) [inline internal/credentials/ecr/pod_identity.go:21]: Is this true for government and AWS (Top) Secret cloud environments?

Comment (krancour) [reply internal/credentials/ecr/pod_identity.go:21]: Good question. After a bit of searching, I am no closer to any clarity on this. I think if this doesn't hold true for those scenarios, we'll hear about it soon enough.

File: internal/credentials/ecr/pod_identity.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	matches := ecrURLRegex.FindStringSubmatch(repoURL)
+	if len(matches) != 3 { // This doesn't look like an ECR URL
+		return "", "", nil
+	}
+	// TODO: We actually might not want to get the account ID from the repoURL
+	// because the account ID in the repoURL may be for a different account from
+	// the one containing the Kargo controller's IAM role and the Project-specific
+	// IAM roles it assumes. (Access across accounts IS possible. It is just not
+	// clear to me yet where else I can get the correct account ID from without
+	// requiring it to be explicitly configured at install-time.)
```
Comment (krancour) [inline internal/credentials/ecr/pod_identity.go:83]: This will be fixed in a follow-up.

**Pull request merged**